--- a/xbmc/cores/VideoPlayer/VideoPlayer.cpp	2017-05-25 08:49:32.000000000 +1200
+++ b/xbmc/cores/VideoPlayer/VideoPlayer.cpp	2017-06-06 16:29:27.887185202 +1200
@@ -3784,7 +3784,19 @@
   {
     if (CSettings::GetInstance().GetInt(CSettings::SETTING_VIDEOPLAYER_ADJUSTREFRESHRATE) != ADJUST_REFRESHRATE_OFF)
     {
-      float framerate = DVD_TIME_BASE / CDVDCodecUtils::NormalizeFrameduration((double)DVD_TIME_BASE * hint.fpsscale / hint.fpsrate);
+      float framerate;
+ 
+      if(hint.fpsscale == 21 && hint.fpsrate == 500)
+      {
+        framerate = 23.976024;
+      }
+      else
+      {
+        framerate = DVD_TIME_BASE / CDVDCodecUtils::NormalizeFrameduration((double)DVD_TIME_BASE * hint.fpsscale / hint.fpsrate);
+      }
+
+      CLog::Log(LOGNOTICE, "VideoPlayer: CVideoPlayer::OpenVideoStream - framerate = %f.", framerate);
+
       m_renderManager.TriggerUpdateResolution(framerate, hint.width, RenderManager::GetStereoModeFlags(hint.stereo_mode));
     }
   }
